src/modules/price-router/Extensions/Balancer/BalancerStablePoolExtension.sol

```diff
@@ -116,13 +113,8 @@ contract BalancerStablePoolExtension is BalancerPoolExtension {
             // Break when a zero address is found.
             if (address(stor.underlyingOrConstituent[i]) == address(0)) break;
             uint256 price = priceRouter.getPriceInUSD(stor.underlyingOrConstituent[i]);
-            if (stor.functionSelectorToGetRate[i] != bytes4(0)) {
-                address constituent = address(stor.underlyingOrConstituent[i]);
-                bytes memory result = constituent.functionStaticCall(
-                    abi.encodeWithSelector(stor.functionSelectorToGetRate[i])
-                );
-                if (result.length == 0) revert BalancerStablePoolExtension__RateProviderCallFailed();
-                uint256 rate = abi.decode(result, (uint256));
+            if (stor.rateProviders[i] != address(0)) {
+                uint256 rate = IRateProvider(stor.rateProviders[i]).getRate();
```

Comment:
We discussed if it is worthwhile to check that the `getRate()` return rate is within an acceptable range. This seems like a highly unlikely scenario that is out of scope for this PR because it kind of applies to general pricing risks, but we will open an issue.

One concept to kickoff this discussion in an issue: Pausing any cellars assuming the multisig can, then work with Strategists to rebalance out of it somehow in a way to mitigate losses.